docs/modules/ROOT/pages/cops_rspec.adoc

```diff
@@ -5493,6 +5493,41 @@ spec/models/user.rb       # shared_examples_for 'foo'
 
 * https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/SpecFilePathSuffix
 
+== RSpec/StringAsInstanceDoubleConstant
+
+|===
+| Enabled by default | Safe | Supports autocorrection | Version Added | Version Changed
+
+| Pending
+| No
+| Always (Unsafe)
+| <<next>>
+| -
+|===
+
+Do not use a string as `instance_double` constant.
+
+=== Safety
+
+This cop is unsafe because the correction requires loading the class.
+Loading before stubbing causes RSpec to only allow instance methods
+to be stubbed.
+
+=== Examples
+
+[source,ruby]
+----
+# bad
+instance_double('User', name: 'John')
+
+# good
+instance_double(User, name: 'John')
```

Comment:
Hi!
Nice catch, we’ve missed this completely.

The new cop is essentially the old one, but with no option to select the style. And this is certainly the direction we choose to enforce.

Your input is valuable, why did you decide to turn the cop off?

In a nutshell, we’ve discovered that an instance_double with a string argument does not verify, and behaves identically to a regular `double`. It makes sense to make the usage explicit, and to use a `double` for abstract doubles, and `instant_double` for specific ones, and verify them.

Reply:
> why did you decide to turn the cop off

We didn't totally turn off `VerifiedDoubleReference`, just ignored a number of existing offences via `.rubocop_todo.yml` — it's a large, old codebase, but Rubocop was only added semi-recently. It's definitely a useful cop for the reasons you've outlined!

That being said, if this new cop is wholly redundant, then I'll likely turn it off 😅. Would it make sense to deprecate one or the other? For what it's worth, `VerifiedDoubleReference` [appears to support](https://github.com/rubocop/rubocop-rspec/blob/a95f2f0d7e785176cbcfea9e70c7fd2fa3d80dfd/lib/rubocop/cop/rspec/verified_double_reference.rb#L50) e.g. `class_double` as well, which I see was a concern raised in this PR.

Reply:
Thanks for sharing.

Yes, it would make sense to deprecate one of the cops, and I’d remove the EnforcedStyle: string option from the old one if it is less intrusive.

Reply:
Ah, thank you for immediately reporting this Luc!

I'm happy to deprecate the new cop and focus on updating the original cop. 

So the action items are 
1. deprecate StringAsInstanceDoubleConstant -- is there a way to make it an alias of the other cop?
2. remove String style from `VerifiedDoubleReference` as it is not consistently verifying. Is there a way to deprecate this or do we simply remove it?

I'll also take a look if there's anything in the new implementation worth porting over to [VerifiedDoubleReference](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecverifieddoublereference

I can probably get started on this on Monday. If anyone wants to pitch in or get started earlier, you are by all means welcome to.

Reply:
I guess we could have it inherit from `VerifiedDoubleReference ` and add the deprecation message, if that makes sense?